File: src/Website/Models/Isaac/Video.cs
using System;
using System.Collections.Generic;

namespace Website.Models.Isaac
{
    public class Video
    {
        public string Id { get; set; }
        public string Title { get; set; }
        public List<Thumbnail> Thumbnails { get; set; }
        public bool RequiresUpdate { get; set; }
        public int Likes { get; set; }
        public int Dislikes { get; set; }
        public int ViewCount { get; set; }
        public int FavouriteCount { get; set; }
        public int CommentCount { get; set; }
        public DateTime Published { get; set; }
        public TimeSpan Duration { get; set; }
        public List<SubmittedEpisode> Submissions { get; set; }
    }
}

File: src/Website/Ts/Models/video.ts
import { Thumbnail } from "./thumbnail";
import { SubmittedEpisode } from "./submitted-episode";

export interface Video {
    id: string,
    title: string,
    published: string,
    duration: string,
    thumbnails: Array<Thumbnail>,
    requires_update: boolean,
    likes: number | null,
    dislikes: number | null,
    view_count: number | null,
    favorite_count: number | null,
    comment_count: number | null,
    submissions: Array<SubmittedEpisode>,
    tags: Array<string>,
    is_3d: boolean,
    is_hd: boolean,
    cc: boolean,
    submission_count: number,
    ratio: number | null,
}